docs: reduce prominence of enterprise notes (#457)
All-Hands-AI/docs: overview/contributing.mdx

### Pull Request Process
We welcome pull requests across our public repositories! Here's how we evaluate them:

> Note: we can't accept pull requests for changes in the `enterprise/` directory of the OpenHands repository right now, since that part of the codebase is commercially licensed. If you have feedback or suggestions for [OpenHands Enterprise](/enterprise/index), please [create an issue](https://github.com/OpenHands/OpenHands/issues) in the OpenHands repository instead.

Author: ```suggestion
Note: we can't accept pull requests for changes in the `enterprise/` directory of the OpenHands repository right now, since that part of the codebase has a separate license. If you have feedback or suggestions for [OpenHands Enterprise](/enterprise/index), please [create an issue](https://github.com/OpenHands/OpenHands/issues) in the OpenHands repository instead.
```

*Full license text: [LICENSE](https://github.com/OpenHands/OpenHands/blob/main/LICENSE)*

> Content in the `enterprise/` directory has a separate license, so we can't accept external pull requests for changes there right now. See `enterprise/LICENSE` for details.

Author: ```suggestion
Content in the `enterprise/` directory has a separate license. See `enterprise/LICENSE` for details.
```
This section is only saying what the license is, duplicating the note here seems a bit less useful maybe?
